File: packages/sdk-python/bidstack_crm/resources/accounts.py
"""Accounts resource."""
from __future__ import annotations
from typing import Any
from ._base import BaseResource


class AccountsResource(BaseResource):
    def list(self, *, page: int = 1, per_page: int = 50) -> dict[str, Any]:
        return self._get("/accounts", params={"page": page, "perPage": per_page})

    def get(self, account_id: str) -> dict[str, Any]:
        return self._get(f"/accounts/{account_id}")

    def create(self, data: dict[str, Any], idempotency_key: str | None = None) -> dict[str, Any]:
        return self._post("/accounts", json=data, idempotency_key=idempotency_key)

    def update(self, account_id: str, data: dict[str, Any]) -> dict[str, Any]:
        return self._patch(f"/accounts/{account_id}", json=data)

    def delete(self, account_id: str) -> None:
        self._delete(f"/accounts/{account_id}")

File: packages/sdk-ruby/lib/bidstack_crm/resources/accounts.rb
require_relative "base"

module BidstackCrm
  module Resources
    class Accounts < Base
      def list(page: 1, per_page: 50)
        get("/accounts", params: { page: page, perPage: per_page })
      end

      def find(account_id) = get("/accounts/#{account_id}")
      def create(data, idempotency_key: nil) = post("/accounts", body: data, idempotency_key: idempotency_key)
      def update(account_id, data) = patch("/accounts/#{account_id}", body: data)
      def destroy(account_id) = delete("/accounts/#{account_id}")
    end
  end
end
